p4/aggregate/aggregate.p4

```diff
@@ -57,14 +52,24 @@ control TpsAggIngress(inout headers hdr,
         default_action = NoAction();
     }
 
-    action data_inspect_packet(bit<32> device) {
-        hdr.gw_int.setValid();
+    action data_inspect_packet(bit<32> device, bit<32> switch_id) {
+        hdr.int_meta.setValid();
+
+        hdr.int_header.remaining_hop_cnt = hdr.int_header.remaining_hop_cnt - 1;
+
+        /* TODO - Find a better means of increasing these sizes using the hdr.int_meta size value */
+        hdr.ipv4.totalLen = hdr.ipv4.totalLen + 12;
+        hdr.int_shim.length = hdr.int_shim.length + 12;
```

Comment:
I think that this should be +3. The length should be in 4-byte words.  This is calculating the length in bytes.